nucleus/utils.py

```diff
@@ -422,3 +425,55 @@ def paginate_generator(
             yield json_value
         if not next_token:
             break
+
+
+def get_image_dimension(image_fpath: str) -> Tuple[int, int]:
+    im = Image.open(image_fpath)
+    return im.size
+
+
+GLOB_SIZE_THRESHOLD_CHECK = 500
+
+
+def create_items_from_folder_crawl(
```

Comment:
Maybe split this method into two? One part would be responsible for globbing, the other could just expect a list of paths and handle the datasetitem creation from them. Maybe we can reuse the second one in other cases as well

Reply:
A bit tricky to create the 2nd function as it has some custom logic for privacy mode.
But I refactored the first part into its own function now.